include/verilated_types.h

```diff
@@ -539,9 +539,27 @@ template <class T_Key, class T_Value> class VlAssocArray final {
 
 private:
     // MEMBERS
-    Map m_map;  // State of the assoc array
+    mutable Map m_map;  // State of the assoc array
     T_Value m_defaultValue;  // Default value
 
+    std::vector<std::pair<std::size_t, T_Value>> modify_input;
+    mutable bool modified = false;
```

Comment:
The `modified` flag seems redundant to me, isn't it always such that `modified == !modify_input.empty()`? If so and there is no appreciable performance difference, then please remove it.